base/abstractset.jl

```diff
@@ -476,6 +476,30 @@ function isdisjoint(a, b)
     _isdisjoint(a, b)
 end
 
+function isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T<:Union{Real,Char}
+    (isempty(a) || isempty(b)) && return true
+    fa, la = first(a), last(a)
+    fb, lb = first(b), last(b)
+    if step(a) < 0
+        fa, la = la, fa
+    end
+    if step(b) < 0
+        fb, lb = lb, fb
+    end
+    not_overlapping = ((la < fb) & (fa < fb)) | ((lb < fa) & (fb < fa))
+    if not_overlapping
+        return not_overlapping
+    elseif abs(step(a)) === abs(step(b))
+        if mod(abs(fa - fb), step(a)) == 0
```

Comment:
floating point is scary. Are you sure this is correct?

Reply:
Agreed! I am fairly sure this is correct, but I don't know how this interacts with the TwicePrecision in numeric ranges. I'm open to suggestions for some way to test and prove/disprove this.

Reply:
Alternatively, I could just cut that branch and more conservatively invoke the unspecialized/slow-path.

Reply:
I think you can use the more reliable check of `if fa in b || la in b || fb in a return false` which is definitely correct. @timholy is the converse true? It should be, but I'm not sure how much I trust that.

Reply:
I think `la in b` is redundant. For integers and other well-behaved types `if fa in b || fb in a return false` should be enough.

Reply:
I think `la in b` is redundant. For integers and other well-behaved types if `fa in b || fb in a return false` should be enough.